diambra/arena/arena_gym.py

```diff
@@ -130,411 +170,176 @@ def render(self, mode='human', wait_key=1):
 
-# DIAMBRA Gym base class providing frame and additional info as observations
-class DiambraGym1P(DiambraGymHardcore1P):
+# DIAMBRA Gym 2P Class
+class DiambraGym2P(DiambraGymBase):
     def __init__(self, env_settings):
         super().__init__(env_settings)
 
         # Dictionary observation space
         observation_space_dict = {}
-        observation_space_dict['frame'] = spaces.Box(low=0, high=255,
-                                                     shape=(self.hwc_dim[0],
-                                                            self.hwc_dim[1],
-                                                            self.hwc_dim[2]),
+        observation_space_dict['frame'] = gym.spaces.Box(low=0, high=255,
+                                                     shape=(self.env_info.frame_shape.h,
+                                                            self.env_info.frame_shape.w,
+                                                            self.env_info.frame_shape.c),
                                                      dtype=np.uint8)
-        player_spec_dict = {}
-
-        # Adding env additional observations (side-specific)
-        for k, v in self.ram_states.items():
 
-            if k == "stage":
-                continue
+        generic_obs_dict, player_obs_dict = self._get_ram_states_obs_dict()
+        observation_space_dict.update(generic_obs_dict)
+        observation_space_dict["agent_0"] = gym.spaces.Dict(player_obs_dict)
+        observation_space_dict["agent_1"] = gym.spaces.Dict(player_obs_dict)
 
-            if k[-2:] == "P1":
-                knew = "own" + k[:-2]
-            else:
-                knew = "opp" + k[:-2]
+        self.observation_space = gym.spaces.Dict(observation_space_dict)
 
-            # Discrete spaces (binary / categorical)
-            if v[0] == 0 or v[0] == 2:
-                player_spec_dict[knew] = spaces.Discrete(v[2] + 1)
-            elif v[0] == 1:  # Box spaces
-                player_spec_dict[knew] = spaces.Box(low=v[1], high=v[2],
-                                                    shape=(1,), dtype=np.int32)
-            else:
-                raise RuntimeError(
-                    "Only Discrete (Binary/Categorical) | Box Spaces allowed")
-
-        actions_dict = {
-            "move": spaces.Discrete(self.n_actions[0]),
-            "attack": spaces.Discrete(self.n_actions[1])
-        }
-
-        player_spec_dict["actions"] = spaces.Dict(actions_dict)
-        observation_space_dict["P1"] = spaces.Dict(player_spec_dict)
-        observation_space_dict["stage"] = spaces.Box(low=self.ram_states["stage"][1],
-                                                     high=self.ram_states["stage"][2],
-                                                     shape=(1,), dtype=np.int8)
-
-        self.observation_space = spaces.Dict(observation_space_dict)
+        # Action space
+        # Dictionary
+        action_space_dict = {}
+        for idx in range(2):
```

Comment:
Why not `for idx, action_space in env_settings.action_space`?

Reply:
Yes, centralized. Let me know how it looks now (and if ok, resolve this)